### PR TITLE
Added initial kml layer support

### PR DESCRIPTION
**Pull request by harrjdk**
Submitting for feedback. I had a need for a static KML Layer so I made the bare-bones for this. I know KML is out of fashion, but I figure this may save some people time moving forward who need KML layers.

### CODE REVIEW

**Comment by diegoazh** [comment]
Could you please fix this error?
```console
/home/travis/build/xkjyeah/vue-google-maps/src/components/kmlLayer.js
  36:1  error  Trailing spaces not allowed  no-trailing-spaces
```

**Comment by diegoazh** [comment]
Please @hornetshell  can you fix the linter error in order to pass the build of this PR? I think that fixing that this work is ready to merge 😃

**Comment by harrjdk** [comment]
Sorry about that! Resolved.

**Comment by diegoazh** [comment]
@hornetshell please can you review your code? one tests fails on CI
```console
Failed tests:
  2) Basic tests Panning of map works:
      Evaluation failed: TypeError: Cannot read property 'lat' of undefined
    at __puppeteer_evaluation_script__:4:24
      at ExecutionContext._evaluateInternal (/home/travis/build/xkjyeah/vue-google-maps/node_modules/puppeteer/lib/ExecutionContext.js:122:13)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:182:7)
    -- ASYNC --
      at ExecutionContext.<anonymous> (/home/travis/build/xkjyeah/vue-google-maps/node_modules/puppeteer/lib/helper.js:111:15)
      at DOMWorld.evaluate (/home/travis/build/xkjyeah/vue-google-maps/node_modules/puppeteer/lib/DOMWorld.js:112:20)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:182:7)
    -- ASYNC --
      at Frame.<anonymous> (/home/travis/build/xkjyeah/vue-google-maps/node_modules/puppeteer/lib/helper.js:111:15)
      at Page.evaluate (/home/travis/build/xkjyeah/vue-google-maps/node_modules/puppeteer/lib/Page.js:782:43)
      at Page.<anonymous> (/home/travis/build/xkjyeah/vue-google-maps/node_modules/puppeteer/lib/helper.js:112:23)
      at evaluate (/home/travis/build/xkjyeah/vue-google-maps/test/test/basic.js:76:37)
      at tryCatch (/home/travis/build/xkjyeah/vue-google-maps/node_modules/regenerator-runtime/runtime.js:45:40)
      at Generator.invoke [as _invoke] (/home/travis/build/xkjyeah/vue-google-maps/node_modules/regenerator-runtime/runtime.js:271:22)
      at Generator.prototype.(anonymous function) [as next] (/home/travis/build/xkjyeah/vue-google-maps/node_modules/regenerator-runtime/runtime.js:97:21)
      at asyncGeneratorStep (/home/travis/build/xkjyeah/vue-google-maps/test/basic.js:50:103)
      at _next (/home/travis/build/xkjyeah/vue-google-maps/test/basic.js:52:194)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:182:7)
1 of 7 tests failed
Test duration: 46732 ms
```

**Comment by diegoazh** [comment]
@hornetshell resolve the requested change below and please execute `npm run test` on your machine print the console and post on a comment in order to merge this PR with trust. This tests is failing on many PRs and only on CI environment. I working on it to find the reason of this fail.
Thank you for your contribution and your work 😃

**Review by diegoazh — CHANGES_REQUESTED** [review]
Please remove the changes introduced on `dist` directory and add a minor upgrade of version on `package.json`.

**Comment by harrjdk** [comment]
I'm looking into the npm tests now. The issue seems to be in the autocomplete example, when I dig through the log. See attached.
[test.log](https://github.com/xkjyeah/vue-google-maps/files/3722607/test.log)

File: src/components/kmlLayer.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ *
+ * KML Layer class (experimental)
+ */
+export default mapElementFactory({
+  mappedProps: props,
+  events,
+  name: 'kmlLayer',
+  ctr: () => google.maps.KmlLayer,
+
+  inject: {
```
Comment (diegoazh) [inline src/components/kmlLayer.js]: Can you remove all this empty options?

File: package-lock.json [hunk 2]
```diff
@@ -5556,30 +5556,6 @@
       "integrity": "sha512-jf84uxzwiuiIVKiOLpfYk7N46TSy8ubTonmneY9vrpHNAnp0QBt2BxWV9dO3/j+BoVAb+a5G6YDPW3M5HOdMWQ==",
       "dev": true
     },
-    "mini-css-extract-plugin": {
```
Comment (diegoazh) [inline package-lock.json:5559]: Why this change?

Comment (harrjdk) [reply package-lock.json:5559]: Package lock was updated to reflect commit b7f90c265f530ee00bc12d9d6d0ce0fa6d4d9861 removing the plugin from package.json.

File: src/components/kmlLayer.js [hunk 3]
```diff
@@ -32,10 +32,6 @@ export default mapElementFactory({
   name: 'kmlLayer',
   ctr: () => google.maps.KmlLayer,
 
-  inject: {
-
-  },
-
   render (h) {
```
Comment (diegoazh) [inline src/components/kmlLayer.js]: @hornetshell sorry, I surelly wasn't clear. I refered to all empty methods
```js
    // no-op
  },

  destroyed () {
    // no-op
  },

  beforeCreate (options) {
    // no-op
  },

  afterCreate (inst) {
    // no-op
  }
```
just left what the component needs.

**Comment by diegoazh** [comment]
Excelent work!!!

**Comment by diegoazh** [comment]
@hornetshell resolve the comments (click on resolve conversation button) to merge this PR

**Comment by diegoazh** [comment]
@hornetshell only one test pass instead of only one test fails can you check that?

**Comment by diegoazh** [comment]
Sorry is the same tests, this work is ok 🎉
